### DISCUSSION TITLE
Code field doesn't escape HTML.

### DISCUSSION THREAD

**Discussion by virtualSpoof** [payloadcms/payload#927] [Feature Requests & Ideas]
# Bug Report
The code field doesn't escape HTML, instead treating it as a HTML to be rendered.

## Expected Behavior
HTML is escaped. 

## Current Behavior
HTML is not escaped.
![image](https://user-images.githubusercontent.com/110639374/182964647-a34474bd-034f-418d-9315-1c15b178ec4a.png)

## Possible Solution
I believe this is due to a 'custom' React state addition, normally React would escape a state value. https://github.com/payloadcms/payload/blob/28e39280943154fd98b3ca9c40a0c26a9488d7cc/src/admin/components/forms/field-types/Code/Code.tsx#L53

## Detailed Description
Latest version. No addons / plugins. Standard.

**Comment by jmikrut** [comment]
Hey @virtualSpoof — ok, I checked into this. I believe that what is necessary here is to extend the `code` field with HTML support (right now, we only support JS, but we could easily support HTML as well with a quick PR).

https://payloadcms.com/docs/fields/code#admin-config

Because this is more of a feature request than a bug, as we know that we only support JS at this time, I'm gonna move this to a discussion. 

PS - I know I personally asked you to create this issue 👼 .  Upon looking into the underlying causes, I realized this is less of a bug and more of a feature. Do you think you'd be up for taking a crack at a PR here? To add HTML support to the Code field?

**Reply by willviles** [reply]
I hit a snag with the syntax highlighting when I wanted to use the code field to: 

- Provide additional freeform json to a collection entity.
- Allow people to write custom JSX components from the CMS

As such, I've opened PR https://github.com/payloadcms/payload/pull/961 to add support for a range of commonly used frontend languages.
